addons/rook/1.4.3/install.sh

```diff
@@ -5,6 +5,10 @@ function rook_pre_init() {
     if [ -n "$version" ] && [ "$version" != "1.4.3" ]; then
         printf "Rook $version is already installed, will not upgrade to 1.4.3\n"
         export SKIP_ROOK_INSTALL='true'
+
+        if [ "$version" = "1.0.4" ] && [ "$KUBERNETES_TARGET_VERSION_MINOR" -ge 20 ]; then
+            bail "Rook 1.0.4 is not compatible with Kubernetes 1.20+"
```

Comment:
why bail here? what if the user wants to upgrade other addons?

Reply:
There's risk that the upgraded version of the add-on requires the upgraded version of Kubernetes. The ticket is ambiguous though:
> when it's in spec: new install - bail and error; existing install - skip update of those components and error

Reply:
The bail suggestion was in regards to a fresh install vs an existing installation upgrade. On a fresh install we didn't want to proceed only to leave them in a non-functional or partially installed state.